server/jetstream_sourcing_scaling_test.go

```diff
@@ -345,45 +433,83 @@ func TestStreamSourcingScalingSourcingManyBenchmark(t *testing.T) {
 		DiscardNewPerSubject: false,
 	})
 	require_NoError(t, err)
-
+	c.waitOnStreamLeader(globalAccountName, "sourcing")
 	sl := c.streamLeader(globalAccountName, "sourcing")
-	mset, err := sl.GlobalAccount().lookupStream("sourcing")
-	require_NoError(t, err)
+	fmt.Printf("Sourcing stream created leader is *** %s ***\n", sl.Name())
 
-	start := time.Now()
+	expectedSeq := make([]int, numSourced)
+
+	//var steppedDown bool
 
-	fmt.Printf("Sourcing stream created\n")
+	start := time.Now()
 
 	var lastMsgs uint64
-	var steppedDown bool
+	mset, err := sl.GlobalAccount().lookupStream("sourcing")
+	require_NoError(t, err)
 
-	checkFor(t, 10*time.Minute, 1*time.Second, func() error {
+	checkFor(t, 5*time.Minute, 1000*time.Millisecond, func() error {
 		mset.mu.RLock()
 		var state StreamState
 		mset.store.FastState(&state)
 		mset.mu.RUnlock()
 
-		if state.Msgs == numMsgPerSource*uint64(numSourced) {
-			fmt.Printf("👍 Test passed: expected %d messages, got %d and took %v\n", numMsgPerSource*uint64(numSourced), state.Msgs, time.Since(start))
+		if state.Msgs == uint64(numMsgPerSource*numSourced) {
+			fmt.Printf("👍 Test passed: expected %d messages, got %d and took %v\n", uint64(numMsgPerSource*numSourced), state.Msgs, time.Since(start))
 			return nil
-		} else if state.Msgs < numMsgPerSource*uint64(numSourced) {
-			fmt.Printf("Current Rate %d/s - Received %d\n", state.Msgs-lastMsgs, state.Msgs)
+		} else if state.Msgs < uint64(numMsgPerSource*numSourced) {
+			fmt.Printf("Current Rate %d per second - Received %d\n", state.Msgs-lastMsgs, state.Msgs)
 			lastMsgs = state.Msgs
 
-			if !steppedDown && state.Msgs >= 200_000 {
-				fmt.Printf("\nStepping Down\n")
-				mset.node.StepDown()
-				steppedDown = true
-				// Need to repull our data.
-				c.waitOnStreamLeader(globalAccountName, "sourcing")
-				sl = c.streamLeader(globalAccountName, "sourcing")
-				mset, err = sl.GlobalAccount().lookupStream("sourcing")
-				require_NoError(t, err)
-			}
-			return fmt.Errorf("Expected %d messages, got %d", numMsgPerSource*uint64(numSourced), state.Msgs)
+			//if !steppedDown && state.Msgs >= 200_000 {
+			//	fmt.Printf("\nStepping Down\n")
+			//	mset.node.StepDown()
+			//	steppedDown = true
+			//	// Need to repull our data.
```

Comment:
commented section here

Reply:
Cleaned up. Those were part of a previous change to the test that would artificially trigger a leader change towards the start of the test, which actually would hide some of the possible failure scenarios.